components/server-side-render/index.js

```diff
@@ -21,45 +28,64 @@ export class ServerSideRender extends Component {
 	}
 
 	componentDidMount() {
+		this.isStillMounted = true;
 		this.fetch( this.props );
 	}
 
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
 	componentWillReceiveProps( nextProps ) {
 		if ( ! isEqual( nextProps, this.props ) ) {
 			this.fetch( nextProps );
 		}
 	}
 
 	fetch( props ) {
-		this.setState( { response: null } );
+		if ( null !== this.state.response ) {
+			this.setState( { response: null } );
+		}
 		const { block, attributes } = props;
 
 		const path = '/gutenberg/v1/block-renderer/' + block + '?context=edit&' + this.getQueryUrlFromObject( { attributes } );
 
-		return wp.apiRequest( { path: path } ).then( ( response ) => {
-			if ( response && response.rendered ) {
+		return apiRequest( { path } ).fail( ( response ) => {
+			const failResponse = {
+				error: true,
+				errorMsg: response.responseJSON.message || __( 'Unknown error' ),
+			};
+			if ( this.isStillMounted ) {
+				this.setState( { response: failResponse } );
+			}
+		} ).done( ( response ) => {
+			if ( this.isStillMounted && response && response.rendered ) {
 				this.setState( { response: response.rendered } );
 			}
 		} );
 	}
 
 	getQueryUrlFromObject( obj, prefix ) {
 		return map( obj, ( paramValue, paramName ) => {
-			const key = prefix ? prefix + '[' + paramName + ']' : paramName,
-				value = obj[ paramName ];
-			return isObject( paramValue ) ? this.getQueryUrlFromObject( value, key ) :
-				encodeURIComponent( key ) + '=' + encodeURIComponent( value );
+			const key = prefix ? prefix + '[' + paramName + ']' : paramName;
+			return isPlainObject( paramValue ) ? this.getQueryUrlFromObject( paramValue, key ) :
+				encodeURIComponent( key ) + '=' + encodeURIComponent( paramValue );
 		} ).join( '&' );
 	}
 
 	render() {
 		const response = this.state.response;
-		if ( ! response || ! response.length ) {
+		if ( ! response ) {
 			return (
-				<div key="loading" className="wp-block-embed is-loading">
-
-					<p>{ __( 'Loading…' ) }</p>
-				</div>
+				<Placeholder><Spinner /></Placeholder>
+			);
+		} else if ( response.error ) {
+			return (
+				<Placeholder>{ __( 'Error loading block: ' ) + response.errorMsg }</Placeholder>
```

Comment:
We'll need to use `wp.i18n.sprintf` here so that this string is translatable into RTL languages like Hebrew.

```js
<Placeholder>{ sprintf( __( 'Error loading block: %s' ), response.errorMsg ) }</Placeholder>
```

Reply:
Thanks for spotting that, added within dcb8048.